packages/openapi-v3/src/json-to-schema.ts

```diff
@@ -32,9 +37,25 @@ export type SchemaRef = ReferenceObject & {definitions: SchemasObject};
 /**
  * Converts JSON Schemas into a SchemaObject
  * @param json - JSON Schema to convert from
+ * @param visited - A map to keep track of mapped json schemas to handle
+ * circular references
  */
-export function jsonToSchemaObject(json: JsonSchema): SchemaObject | SchemaRef {
-  const result: SchemaObject = {};
+export function jsonToSchemaObject(
+  json: JsonSchema,
+  visited: Map<JsonSchema, SchemaObject | SchemaRef> = new Map(),
+): SchemaObject | SchemaRef {
+  // A flag to check if a schema object is fully converted
+  const converted = 'x-loopback-converted';
+  const schema = visited.get(json);
+  if (schema != null && isSchemaObject(schema) && schema[converted] === false) {
+    return {$ref: `#/components/schemas/${json.title}`};
+  }
+  if (schema != null) return schema;
+
+  const result: SchemaObject = {
+    [converted]: false,
```

Comment:
'x-loopback-converted' is never set to true. It is just a placeholder to verify the result object. why not directly declare it as a field here ?

Reply:
`x-*` is extensibility elements for OpenAPI specs.